lib/hpke/include/hpke/signature.h

```diff
@@ -21,6 +21,7 @@ struct Signature
   template<Signature::ID id>
   static const Signature& get();
 
+  Signature(ID id_in);
   virtual ~Signature() = default;
```

Comment:
This allows Signature to be created in 2 different ways. One via this constructor and one a concrete version via get<T>() API.

Reply:
Good catch, moved those ctors to be protected, since they should only be called by subclasses.